experimental/devices/rainbowhat/rainbowhat.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package rainbowhat
+
+import (
+	"periph.io/x/periph/conn/spi/spireg"
+	"periph.io/x/periph/devices/apa102"
+	"periph.io/x/periph/devices/bmxx80"
+	"periph.io/x/periph/experimental/devices/ht16k33"
+	"periph.io/x/periph/host/rpi"
+
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/i2c/i2creg"
+)
+
+const (
+	numPixels  = 7
+	bmp280Addr = 0x77
+)
+
+// Dev represents a Rainbow HAT  (https://shop.pimoroni.com/products/rainbow-hat-for-android-things)
+type Dev struct {
+	ledstrip *apa102.Dev
+	bmp280   *bmxx80.Dev
+	display  *ht16k33.Display
+	buttonA  gpio.PinIn
+	buttonB  gpio.PinIn
+	buttonC  gpio.PinIn
+	ledR     gpio.PinOut
+	ledG     gpio.PinOut
+	ledB     gpio.PinOut
+	buzzer   gpio.PinOut
+	servo    gpio.PinOut
+}
+
+// NewRainbowHat returns a rainbowhat driver.
+func NewRainbowHat() (*Dev, error) {
+	i2cPort, err := i2creg.Open("")
+	if err != nil {
+		return nil, err
+	}
+
+	spiPort, err := spireg.Open("")
+	if err != nil {
+		return nil, err
+	}
+
+	bmp280, err := bmxx80.NewI2C(i2cPort, bmp280Addr, &bmxx80.DefaultOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	display, err := ht16k33.NewAlphaNumericDisplay(i2cPort, ht16k33.I2CAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := apa102.DefaultOpts
```

Comment:
Maybe the user would like to pass options there, so I think it'd be worth having the client to pass this explicitly so they can choose the brightness and temperature. Then make a copy and override the number of pixels, i.e.
```
func NewRainbowHat(ao * apa102.Opts) (*Dev, error) {
  ...
  opts := *ao
  opts.NumPixels = 7
  ...
```

That said .. it's not a big deal either, users can update these values afterward. So thinking about it a bit more, I guess you can ignore this comment.